toolchain/check/testdata/packages/no_prelude/cross_package_export.carbon

```diff
@@ -0,0 +1,429 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+// ============================================================================
+// Setup files
+// ============================================================================
+
+// --- base.carbon
+
+package Other library "base";
+
+class C {
+  var x: ();
+};
+
+// --- export_import.carbon
+
+package Other library "export_import";
+
+export import library "base";
+
+// --- export_import_copy.carbon
+
+package Other library "export_import_copy";
+
+export import library "base";
+
+// --- export_name.carbon
+
+package Other library "export_name";
+
+import library "base";
+
+export C;
+
+// --- export_name_copy.carbon
+
+package Other library "export_name_copy";
+
+import library "base";
+
+export C;
+
+// ============================================================================
+// Test files
+// ============================================================================
+
+// --- fail_todo_use_export_import.carbon
+
+library "use_export_import";
+
+import Other library "export_import";
+
+// CHECK:STDERR: fail_todo_use_export_import.carbon:[[@LINE+4]]:8: ERROR: Name `C` not found.
+// CHECK:STDERR: var c: Other.C = {.x = ()};
+// CHECK:STDERR:        ^~~~~~~
+// CHECK:STDERR:
+var c: Other.C = {.x = ()};
+
+// --- fail_todo_use_export_import_both.carbon
+
+library "use_export_import_both";
+
+import Other library "export_import";
+import Other library "export_import_copy";
+
+// CHECK:STDERR: fail_todo_use_export_import_both.carbon:[[@LINE+4]]:8: ERROR: Name `C` not found.
+// CHECK:STDERR: var c: Other.C = {.x = ()};
+// CHECK:STDERR:        ^~~~~~~
+// CHECK:STDERR:
+var c: Other.C = {.x = ()};
+
+// --- use_export_name.carbon
+
+library "use_export_name";
+
+import Other library "export_name";
+
+var c: Other.C = {.x = ()};
+
+// --- fail_todo_use_export_name_both.carbon
+
+library "use_export_name_both";
+
+import Other library "export_name";
+// CHECK:STDERR: fail_todo_use_export_name_both.carbon:[[@LINE+18]]:1: In import.
+// CHECK:STDERR: import Other library "export_name_copy";
+// CHECK:STDERR: ^~~~~~
+// CHECK:STDERR: export_name_copy.carbon:4:1: In import.
+// CHECK:STDERR: import library "base";
+// CHECK:STDERR: ^~~~~~
+// CHECK:STDERR: base.carbon:4:1: ERROR: Duplicate name being declared in the same scope.
+// CHECK:STDERR: class C {
+// CHECK:STDERR: ^~~~~~~~~
+// CHECK:STDERR: fail_todo_use_export_name_both.carbon:[[@LINE-10]]:1: In import.
+// CHECK:STDERR: import Other library "export_name";
+// CHECK:STDERR: ^~~~~~
```

Comment:
Not about this PR at all, but maybe for future stuff -- it's a bit awkward taht we end up doing the import walk twice here. Is it once for the error and once for the note? If so, I at least have an idea of how this might be addressed.